etc/dbus-serialbattery/battery.py

```diff
@@ -272,12 +278,14 @@ def manage_charge_voltage_linear(self) -> None:
                     ) and not self.allow_max_voltage:
                         self.allow_max_voltage = True
                 else:
-                    tDiff = time() - self.max_voltage_start_time
+                    tDiff = int(time()) - self.max_voltage_start_time
                     # if utils.MAX_VOLTAGE_TIME_SEC < tDiff:
                     # keep max voltage for 300 more seconds
                     if 300 < tDiff:
                         self.allow_max_voltage = False
                         self.max_voltage_start_time = None
+                    if voltageSum < self.max_battery_voltage - utils.VOLTAGE_DROP:
```

Comment:
Please add a comment for what this check is

Reply:
if battery get almost full charged, the charge_mode toggles back and forth  bulk dynamic and absorption dynamic. Activation of absorption (dynamic) starts timer max_voltage_start_time but
does not reset it wenn battery left absorption (dynamic) and goes to bulk (dynamic) 

This means that if the battery jumps back into absorption (dynamic), the cell will still run with the old value. 

If we left absorption (dynamic) ----> voltageSum < self.max_battery_voltage - utils.VOLTAGE_DROP
we always reset  max_voltage_start_time

This debug output illustrated it.

it means 
with # + str(tDiff) in mode 
```
self.charge_mode = (
                    "Bulk dynamic"
                    # + " (vS: "
                    # + str(round(voltageSum, 2))
                    # + " - pS: "
                    # + str(round(penaltySum, 2))
                    # + ")"
                    if self.max_voltage_start_time is None
                    else "Absorption dynamic"
                    # + "(vS: "
                    # + str(round(voltageSum, 2))
                    # + " tDiff: "
                    # + str(tDiff)
                    # + " pS: "
                    # + str(round(penaltySum, 2))
                    # + ")"
                )
```
.....
Resetting self.max_voltage_start_time when voltageSum falls below the maximum: This change addresses a specific scenario where a single cell's voltage exceeds the maximum while the total battery voltage (voltageSum) has not yet reached the maximum. In this case, voltageSum can fluctuate around the threshold value just before the start of Linear Absorption mode. By resetting self.max_voltage_start_time during a voltage drop, the code can more accurately track the start of the Linear Absorption mode.
.....